package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,371 @@
+            """
+        if isinstance(u, mda.Universe):
+            self._dist_matrix = DistanceMatrix(u, **kwargs)
+        elif isinstance(u, DistanceMatrix):
+            self._dist_matrix = u
+        else:
+            raise ValueError("U is not a Universe or DistanceMatrix and"
+                             " so the DiffusionMap has no data to work with.")
+        self._epsilon = epsilon
+        # important for transform function and length of .run() method
+        self._nframes = self._dist_matrix.nframes
+        if self._nframes > 2000:
+            warnings.warn("The distance matrix is very large, and can "
+                          "be very slow to compute. Consider picking a larger "
+                          "step size in distance matrix initialization.")
+
+        # determines length of diffusion process
+        self._t = timescale
+
+        if manifold_density is None:
+            # weights do not apply to metric but density of data
+            self._weights_ker = np.ones((self._nframes,))
+        else:
+            if manifold_density.shape[0] != self._nframes:
+                raise ValueError("The weight should have the same length as "
+                                 'the trajectory')
+            else:
+                # density weights are constructed as relative to the mean
+                self._weights_ker = (np.asarray(manifold_density,
+                                     dtype=np.float64) /
+                                     np.mean(manifold_density))
+
+    def run(self):
+        # will only run if distance matrix not already calculated
+        self._dist_matrix.run()
+        self._scaled_matrix = self._dist_matrix.dist_matrix / self._epsilon
+        # this should be a reference to the same object as
+        # self.dist_matrix.dist_matrix
+        # take negative exponent of scaled matrix to create Isotropic kernel
+        self._kernel = np.exp(-self._scaled_matrix)
+        # define an anistropic diffusion term q
+        q_vector = np.zeros((self._nframes, ))
+        # weights should reflect the density of the points on the manifold
+        for i in range(self._nframes):
+            q_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        # Form a new kernel from the anisotropic diffusion term q
+        self._kernel /= np.sqrt(q_vector[:, np.newaxis].dot(q_vector[np.newaxis]))
+
+        # Weighted Graph Laplacian normalization on this new graph
+        d_vector = np.zeros((self._nframes, ))
+        for i in range(self._nframes):
+            d_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        for i in range(self._nframes):
+            self._kernel[i, :] = self._kernel[i, :] * self._weights_ker
+
+        # Define anisotropic transition by dividing kernel by this term
+        self._kernel /= np.sqrt(d_vector[:, np.newaxis].dot(d_vector[np.newaxis]))
+
+        # Apply timescaling
+        for i in range(self._t):
+            if i > 0:
```

Comment:
`for i in range(1, self._t)`

Another thing. Isn't there a matrix power somewhere in numpy/scipy?